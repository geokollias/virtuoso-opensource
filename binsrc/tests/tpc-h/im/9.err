Starting Impala Shell without Kerberos authentication
Connected to ip-10-0-1-251.ec2.internal:21000
Server version: impalad version 2.1.3-cdh5 RELEASE (build 20816e26a150d20d2c92c470aa40b342521b243e)
Query: use TPCH
Query: select nation,
 o_year,
 sum(amount) as sum_profit
from
 (
  select
   n_name as nation,
   extract(year from o_orderdate) as o_year,
   l_extendedprice * (1 - l_discount) - ps_supplycost * l_quantity as amount
  from
   part,
   supplier,
   lineitem,
   partsupp,
   orders,
   nation
  where
   s_suppkey = l_suppkey
   and ps_suppkey = l_suppkey
   and ps_partkey = l_partkey
   and p_partkey = l_partkey
   and o_orderkey = l_orderkey
   and s_nationkey = n_nationkey
   and p_name like '%green%'
 ) as profit
group by
 nation,
 o_year
order by
 nation,
 o_year desc
Fetched 175 row(s) in 228.79s
