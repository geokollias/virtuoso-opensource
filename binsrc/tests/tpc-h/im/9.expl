[?1034h+------------------------------------------------------------------------------------+
| Explain String                                                                     |
+------------------------------------------------------------------------------------+
| Estimated Per-Host Requirements: Memory=5.85GB VCores=6                            |
| WARNING: The following tables are missing relevant table and/or column statistics. |
| tpch.nation, tpch.part                                                             |
|                                                                                    |
| 20:MERGING-EXCHANGE [UNPARTITIONED]                                                |
| |  order by: nation ASC, o_year DESC                                               |
| |  hosts=1 per-host-mem=unavailable                                                |
| |  tuple-ids=8 row-size=37B cardinality=2706944657                                 |
| |                                                                                  |
| 12:SORT                                                                            |
| |  order by: nation ASC, o_year DESC                                               |
| |  hosts=1 per-host-mem=880.00MB                                                   |
| |  tuple-ids=8 row-size=37B cardinality=2706944657                                 |
| |                                                                                  |
| 19:AGGREGATE [FINALIZE]                                                            |
| |  output: sum:merge(amount)                                                       |
| |  group by: nation, o_year                                                        |
| |  hosts=1 per-host-mem=128.00MB                                                   |
| |  tuple-ids=7 row-size=37B cardinality=2706944657                                 |
| |                                                                                  |
| 18:EXCHANGE [HASH(nation,o_year)]                                                  |
| |  hosts=1 per-host-mem=0B                                                         |
| |  tuple-ids=7 row-size=37B cardinality=2706944657                                 |
| |                                                                                  |
| 11:AGGREGATE                                                                       |
| |  output: sum(l_extendedprice * (1 - l_discount) - ps_supplycost * l_quantity)    |
| |  group by: n_name, EXTRACT(year FROM o_orderdate)                                |
| |  hosts=1 per-host-mem=128.00MB                                                   |
| |  tuple-ids=7 row-size=37B cardinality=2706944657                                 |
| |                                                                                  |
| 10:HASH JOIN [INNER JOIN, BROADCAST]                                               |
| |  hash predicates: l_suppkey = ps_suppkey, l_partkey = ps_partkey                 |
| |  hosts=1 per-host-mem=1.31GB                                                     |
| |  tuple-ids=2,0,1,5,4,3 row-size=137B cardinality=2706944657                      |
| |                                                                                  |
| |--17:EXCHANGE [BROADCAST]                                                         |
| |  |  hosts=1 per-host-mem=0B                                                      |
| |  |  tuple-ids=3 row-size=16B cardinality=80000000                                |
| |  |                                                                               |
| |  03:SCAN HDFS [tpch.partsupp, RANDOM]                                            |
| |     partitions=1/1 files=18 size=4.21GB                                          |
| |     table stats: 80000000 rows total                                             |
| |     column stats: all                                                            |
| |     hosts=1 per-host-mem=264.00MB                                                |
| |     tuple-ids=3 row-size=16B cardinality=80000000                                |
| |                                                                                  |
| 09:HASH JOIN [INNER JOIN, BROADCAST]                                               |
| |  hash predicates: l_orderkey = o_orderkey                                        |
| |  hosts=1 per-host-mem=3.69GB                                                     |
| |  tuple-ids=2,0,1,5,4 row-size=121B cardinality=794272581                         |
| |                                                                                  |
| |--16:EXCHANGE [BROADCAST]                                                         |
| |  |  hosts=1 per-host-mem=0B                                                      |
| |  |  tuple-ids=4 row-size=24B cardinality=150000000                               |
| |  |                                                                               |
| |  04:SCAN HDFS [tpch.orders, RANDOM]                                              |
| |     partitions=1/1 files=25 size=6.02GB                                          |
| |     table stats: 150000000 rows total                                            |
| |     column stats: all                                                            |
| |     hosts=1 per-host-mem=176.00MB                                                |
| |     tuple-ids=4 row-size=24B cardinality=150000000                               |
| |                                                                                  |
| 08:HASH JOIN [INNER JOIN, BROADCAST]                                               |
| |  hash predicates: s_nationkey = n_nationkey                                      |
| |  hosts=1 per-host-mem=798B                                                       |
| |  tuple-ids=2,0,1,5 row-size=97B cardinality=600037902                            |
| |                                                                                  |
| |--15:EXCHANGE [BROADCAST]                                                         |
| |  |  hosts=1 per-host-mem=0B                                                      |
| |  |  tuple-ids=5 row-size=29B cardinality=25                                      |
| |  |                                                                               |
| |  05:SCAN HDFS [tpch.nation, RANDOM]                                              |
| |     partitions=1/1 files=1 size=2.17KB                                           |
| |     table stats: 25 rows total                                                   |
| |     columns missing stats: n_name                                                |
| |     hosts=1 per-host-mem=32.00MB                                                 |
| |     tuple-ids=5 row-size=29B cardinality=25                                      |
| |                                                                                  |
| 07:HASH JOIN [INNER JOIN, BROADCAST]                                               |
| |  hash predicates: l_suppkey = s_suppkey                                          |
| |  hosts=1 per-host-mem=8.39MB                                                     |
| |  tuple-ids=2,0,1 row-size=68B cardinality=600037902                              |
| |                                                                                  |
| |--14:EXCHANGE [BROADCAST]                                                         |
| |  |  hosts=1 per-host-mem=0B                                                      |
| |  |  tuple-ids=1 row-size=8B cardinality=1000000                                  |
| |  |                                                                               |
| |  01:SCAN HDFS [tpch.supplier, RANDOM]                                            |
| |     partitions=1/1 files=1 size=77.32MB                                          |
| |     table stats: 1000000 rows total                                              |
| |     column stats: all                                                            |
| |     hosts=1 per-host-mem=176.00MB                                                |
| |     tuple-ids=1 row-size=8B cardinality=1000000                                  |
| |                                                                                  |
| 06:HASH JOIN [INNER JOIN, BROADCAST]                                               |
| |  hash predicates: l_partkey = p_partkey                                          |
| |  hosts=1 per-host-mem=41.96MB                                                    |
| |  tuple-ids=2,0 row-size=60B cardinality=600037902                                |
| |                                                                                  |
| |--13:EXCHANGE [BROADCAST]                                                         |
| |  |  hosts=1 per-host-mem=0B                                                      |
| |  |  tuple-ids=0 row-size=20B cardinality=2000000                                 |
| |  |                                                                               |
| |  00:SCAN HDFS [tpch.part, RANDOM]                                                |
| |     partitions=1/1 files=3 size=648.75MB                                         |
| |     predicates: p_name LIKE '%green%'                                            |
| |     table stats: 20000000 rows total                                             |
| |     columns missing stats: p_name                                                |
| |     hosts=1 per-host-mem=176.00MB                                                |
| |     tuple-ids=0 row-size=20B cardinality=2000000                                 |
| |                                                                                  |
| 02:SCAN HDFS [tpch.lineitem, RANDOM]                                               |
|    partitions=1/1 files=82 size=20.20GB                                            |
|    table stats: 600037902 rows total                                               |
|    column stats: all                                                               |
|    hosts=1 per-host-mem=528.00MB                                                   |
|    tuple-ids=2 row-size=40B cardinality=600037902                                  |
+------------------------------------------------------------------------------------+
