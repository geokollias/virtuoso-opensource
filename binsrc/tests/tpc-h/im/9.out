[?1034h+---------------------------+--------+-------------------+
| nation                    | o_year | sum_profit        |
+---------------------------+--------+-------------------+
| ALGERIA                   | 1998   | 2746388473.803196 |
| ALGERIA                   | 1997   | 4705096214.193357 |
| ALGERIA                   | 1996   | 4691101028.388028 |
| ALGERIA                   | 1995   | 4692240612.22093  |
| ALGERIA                   | 1994   | 4677512644.204861 |
| ALGERIA                   | 1993   | 4684367583.828131 |
| ALGERIA                   | 1992   | 4704266160.031439 |
| ARGENTINA                 | 1998   | 2718513010.851665 |
| ARGENTINA                 | 1997   | 4646318596.684613 |
| ARGENTINA                 | 1996   | 4656102598.041129 |
| ARGENTINA                 | 1995   | 4652753173.228324 |
| ARGENTINA                 | 1994   | 4645991799.532767 |
| ARGENTINA                 | 1993   | 4643401137.562661 |
| ARGENTINA                 | 1992   | 4617848896.771819 |
| BRAZIL                    | 1998   | 2731140677.116336 |
| BRAZIL                    | 1997   | 4675931143.507588 |
| BRAZIL                    | 1996   | 4665397071.981847 |
| BRAZIL                    | 1995   | 4661653435.263788 |
| BRAZIL                    | 1994   | 4664829056.380774 |
| BRAZIL                    | 1993   | 4675246060.52108  |
| BRAZIL                    | 1992   | 4699378591.616911 |
| CANADA                    | 1998   | 2737663014.185818 |
| CANADA                    | 1997   | 4680387035.72936  |
| CANADA                    | 1996   | 4682077548.127958 |
| CANADA                    | 1995   | 4683532053.525635 |
| CANADA                    | 1994   | 4661595473.324359 |
| CANADA                    | 1993   | 4663422610.270894 |
| CANADA                    | 1992   | 4693236211.670271 |
| CHINA                     | 1998   | 2734194369.805998 |
| CHINA                     | 1997   | 4683510835.725309 |
| CHINA                     | 1996   | 4694563430.790443 |
| CHINA                     | 1995   | 4659549932.033935 |
| CHINA                     | 1994   | 4668649255.314692 |
| CHINA                     | 1993   | 4653663845.278809 |
| CHINA                     | 1992   | 4671594192.392078 |
| EGYPT                     | 1998   | 2730076443.188374 |
| EGYPT                     | 1997   | 4641217395.731147 |
| EGYPT                     | 1996   | 4682023903.334127 |
| EGYPT                     | 1995   | 4656985688.995199 |
| EGYPT                     | 1994   | 4639974596.479136 |
| EGYPT                     | 1993   | 4654217487.634022 |
| EGYPT                     | 1992   | 4655508405.620231 |
| ETHIOPIA                  | 1998   | 2771891798.562638 |
| ETHIOPIA                  | 1997   | 4691462902.651324 |
| ETHIOPIA                  | 1996   | 4699807893.851868 |
| ETHIOPIA                  | 1995   | 4712019051.049996 |
| ETHIOPIA                  | 1994   | 4714485895.714344 |
| ETHIOPIA                  | 1993   | 4673334464.756043 |
| ETHIOPIA                  | 1992   | 4709982012.233111 |
| FRANCE                    | 1998   | 2749352687.477413 |
| FRANCE                    | 1997   | 4691958256.233721 |
| FRANCE                    | 1996   | 4720197405.493249 |
| FRANCE                    | 1995   | 4702056754.286272 |
| FRANCE                    | 1994   | 4701239088.153422 |
| FRANCE                    | 1993   | 4712310290.57282  |
| FRANCE                    | 1992   | 4706154746.094987 |
| GERMANY                   | 1998   | 2767125834.858303 |
| GERMANY                   | 1997   | 4703923545.677734 |
| GERMANY                   | 1996   | 4690487560.374782 |
| GERMANY                   | 1995   | 4686747664.025197 |
| GERMANY                   | 1994   | 4660352911.183455 |
| GERMANY                   | 1993   | 4692160055.611502 |
| GERMANY                   | 1992   | 4691498075.881687 |
| INDIA                     | 1998   | 2732579085.731987 |
| INDIA                     | 1997   | 4678477185.982782 |
| INDIA                     | 1996   | 4673866490.267288 |
| INDIA                     | 1995   | 4674959076.822014 |
| INDIA                     | 1994   | 4681827178.388269 |
| INDIA                     | 1993   | 4643481746.815372 |
| INDIA                     | 1992   | 4665882504.058039 |
| INDONESIA                 | 1998   | 2745112789.930809 |
| INDONESIA                 | 1997   | 4665596134.637309 |
| INDONESIA                 | 1996   | 4669667677.069337 |
| INDONESIA                 | 1995   | 4667166725.642455 |
| INDONESIA                 | 1994   | 4662762228.671452 |
| INDONESIA                 | 1993   | 4656778134.171697 |
| INDONESIA                 | 1992   | 4670973296.65157  |
| IRAN                      | 1998   | 2723216170.395517 |
| IRAN                      | 1997   | 4609684248.849324 |
| IRAN                      | 1996   | 4628753160.933533 |
| IRAN                      | 1995   | 4627120996.279796 |
| IRAN                      | 1994   | 4608487551.028377 |
| IRAN                      | 1993   | 4620764336.354362 |
| IRAN                      | 1992   | 4627902203.820455 |
| IRAQ                      | 1998   | 2740906684.589864 |
| IRAQ                      | 1997   | 4666298134.812037 |
| IRAQ                      | 1996   | 4684322817.760218 |
| IRAQ                      | 1995   | 4653983196.706611 |
| IRAQ                      | 1994   | 4683335111.020104 |
| IRAQ                      | 1993   | 4667160512.266254 |
| IRAQ                      | 1992   | 4678315491.605583 |
| JAPAN                     | 1998   | 2724504496.603639 |
| JAPAN                     | 1997   | 4609764983.846019 |
| JAPAN                     | 1996   | 4650986737.210389 |
| JAPAN                     | 1995   | 4663812104.33335  |
| JAPAN                     | 1994   | 4633301980.70184  |
| JAPAN                     | 1993   | 4638916610.473583 |
| JAPAN                     | 1992   | 4656081490.079933 |
| JORDAN                    | 1998   | 2742880098.57751  |
| JORDAN                    | 1997   | 4672752862.433925 |
| JORDAN                    | 1996   | 4667670128.211616 |
| JORDAN                    | 1995   | 4651272615.681397 |
| JORDAN                    | 1994   | 4670362946.344707 |
| JORDAN                    | 1993   | 4673339129.912154 |
| JORDAN                    | 1992   | 4675813230.028477 |
| KENYA                     | 1998   | 2717580810.999946 |
| KENYA                     | 1997   | 4606427219.57424  |
| KENYA                     | 1996   | 4628365122.130834 |
| KENYA                     | 1995   | 4611166905.220373 |
| KENYA                     | 1994   | 4616660205.305481 |
| KENYA                     | 1993   | 4629037306.568191 |
| KENYA                     | 1992   | 4635940424.656764 |
| MOROCCO                   | 1998   | 2728973692.691986 |
| MOROCCO                   | 1997   | 4677385007.667835 |
| MOROCCO                   | 1996   | 4691138516.359416 |
| MOROCCO                   | 1995   | 4672866595.741712 |
| MOROCCO                   | 1994   | 4679363707.899522 |
| MOROCCO                   | 1993   | 4657343605.486782 |
| MOROCCO                   | 1992   | 4673293158.183667 |
| MOZAMBIQUE                | 1998   | 2746870372.986606 |
| MOZAMBIQUE                | 1997   | 4657797818.025125 |
| MOZAMBIQUE                | 1996   | 4680907763.269811 |
| MOZAMBIQUE                | 1995   | 4679365111.693347 |
| MOZAMBIQUE                | 1994   | 4650089862.929606 |
| MOZAMBIQUE                | 1993   | 4674781658.723163 |
| MOZAMBIQUE                | 1992   | 4703066893.366589 |
| PERU                      | 1998   | 2701947354.748188 |
| PERU                      | 1997   | 4623755865.673775 |
| PERU                      | 1996   | 4653145387.625609 |
| PERU                      | 1995   | 4649971644.393022 |
| PERU                      | 1994   | 4640145803.974966 |
| PERU                      | 1993   | 4630471111.060339 |
| PERU                      | 1992   | 4629977353.555231 |
| ROMANIA                   | 1998   | 2730185251.676894 |
| ROMANIA                   | 1997   | 4662919818.677359 |
| ROMANIA                   | 1996   | 4692623447.604005 |
| ROMANIA                   | 1995   | 4683093538.177529 |
| ROMANIA                   | 1994   | 4656908200.898009 |
| ROMANIA                   | 1993   | 4673109064.641903 |
| ROMANIA                   | 1992   | 4659501236.859735 |
| RUSSIA                    | 1998   | 2738455915.543599 |
| RUSSIA                    | 1997   | 4701455238.060455 |
| RUSSIA                    | 1996   | 4694648504.034442 |
| RUSSIA                    | 1995   | 4674020866.806455 |
| RUSSIA                    | 1994   | 4687989534.597377 |
| RUSSIA                    | 1993   | 4680591337.852927 |
| RUSSIA                    | 1992   | 4684865871.570374 |
| SAUDI ARABIA              | 1998   | 2722192493.211489 |
| SAUDI ARABIA              | 1997   | 4646872112.018315 |
| SAUDI ARABIA              | 1996   | 4679097125.499868 |
| SAUDI ARABIA              | 1995   | 4659365308.242087 |
| SAUDI ARABIA              | 1994   | 4668096847.309839 |
| SAUDI ARABIA              | 1993   | 4647885741.922552 |
| SAUDI ARABIA              | 1992   | 4665645732.088393 |
| UNITED KINGDOM            | 1998   | 2757252921.493515 |
| UNITED KINGDOM            | 1997   | 4687915843.382687 |
| UNITED KINGDOM            | 1996   | 4725596237.952026 |
| UNITED KINGDOM            | 1995   | 4692146767.049665 |
| UNITED KINGDOM            | 1994   | 4717398081.811819 |
| UNITED KINGDOM            | 1993   | 4694183885.294186 |
| UNITED KINGDOM            | 1992   | 4722609721.776423 |
| UNITED STATES             | 1998   | 2744344335.038303 |
| UNITED STATES             | 1997   | 4652698592.586711 |
| UNITED STATES             | 1996   | 4670553955.705688 |
| UNITED STATES             | 1995   | 4659218769.759757 |
| UNITED STATES             | 1994   | 4671044724.080699 |
| UNITED STATES             | 1993   | 4680519402.476569 |
| UNITED STATES             | 1992   | 4707714157.320902 |
| VIETNAM                   | 1998   | 2741387136.246842 |
| VIETNAM                   | 1997   | 4661928954.85413  |
| VIETNAM                   | 1996   | 4690535970.800281 |
| VIETNAM                   | 1995   | 4692078285.128686 |
| VIETNAM                   | 1994   | 4705541885.249335 |
| VIETNAM                   | 1993   | 4708733491.560843 |
| VIETNAM                   | 1992   | 4681498238.758052 |
+---------------------------+--------+-------------------+
